ambry-server/src/test/java/com.github.ambry.server/AmbryRequestsTest.java

```diff
@@ -617,6 +674,37 @@ private void doRequestControlRequestTest(RequestOrResponseType toControl, Partit
     sendAndVerifyOperationRequest(toControl, idsToTest, ServerErrorCode.No_Error);
   }
 
+  /**
+   * Sends and verifies that GetRequest with a list of original blobIds works correctly.
+   * @param blobIds List of blobIds for GetRequest.
+   * @param expectedErrorCode the {@link ServerErrorCode} expected in the response.
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  private void sendAndVerifyGetOriginalStoreKeys(List<BlobId> blobIds, ServerErrorCode expectedErrorCode)
+      throws InterruptedException, IOException {
+    PartitionId partitionId = blobIds.get(0).getPartition();
+    int correlationId = blobIds.get(0).getContainerId();
+    String clientId = UtilsTest.getRandomString(10);
+
+    PartitionRequestInfo pRequestInfo = new PartitionRequestInfo(partitionId, blobIds);
+    RequestOrResponse request =
+        new GetRequest(correlationId, clientId, MessageFormatFlags.All, Collections.singletonList(pRequestInfo),
+            GetOption.Include_All);
+    storageManager.resetStore();
+    if (!expectedErrorCode.equals(ServerErrorCode.Unknown_Error)) {
+      // known error will be filled to each PartitionResponseInfo and set ServerErrorCode.No_Error in response.
+      Response response = sendRequestGetResponse(request, ServerErrorCode.No_Error);
+      assertEquals("Operation received at the store not as expected", RequestOrResponseType.GetRequest,
+          MockStorageManager.operationReceived);
+      for (PartitionResponseInfo info : ((GetResponse) response).getPartitionResponseInfoList()) {
+        assertEquals("Error code does not match expected", expectedErrorCode, info.getErrorCode());
```

Comment:
would it also be useful to check what keys you got in the store? How do you know you got the right converted key?

I'll merge this for now but let's make this change in another PR if required